src/pages/instances/forms/CreateImageFromInstanceSnapshotForm.tsx

```diff
@@ -0,0 +1,122 @@
+import { FC } from "react";
+import { LxdInstance, LxdInstanceSnapshot } from "types/instance";
+import { useEventQueue } from "context/eventQueue";
+import { useFormik } from "formik";
+import { useToastNotification } from "context/toastNotificationProvider";
+import { createImageAlias, createImageFromInstanceSnapshot } from "api/images";
+import {
+  ActionButton,
+  Button,
+  Form,
+  Input,
+  Modal,
+} from "@canonical/react-components";
+import * as Yup from "yup";
+import { Link } from "react-router-dom";
+
+interface Props {
+  instance: LxdInstance;
+  snapshot: LxdInstanceSnapshot;
+  close: () => void;
+}
+
+const CreateImageFromInstanceSnapshotForm: FC<Props> = ({
+  instance,
+  snapshot,
+  close,
+}) => {
+  const eventQueue = useEventQueue();
+  const toastNotify = useToastNotification();
+
+  const notifySuccess = () => {
+    const published = (
+      <Link to={`/ui/project/${instance.project}/images`}>published</Link>
+    );
+    toastNotify.success(<>Image {published}.</>);
+  };
+
+  const formik = useFormik<{ alias: string; isPublic: boolean }>({
+    initialValues: {
+      alias: "",
+      isPublic: false,
+    },
+    validationSchema: Yup.object().shape({
+      alias: Yup.string(),
+    }),
+    onSubmit: (values) => {
+      const alias = values.alias;
+
+      createImageFromInstanceSnapshot(instance, snapshot, values.isPublic)
+        .then((operation) => {
+          toastNotify.info("Publishing of image started.");
```

Comment:
maybe add snapshot name to the info toast message?

Reply:
I improved the wording of the notifications a bit, so the notifications always include the snapshot name.